ext/spl/spl_fixedarray.c

```diff
@@ -107,14 +107,16 @@ static void spl_fixedarray_resize(spl_fixedarray *array, zend_long size) /* {{{
 	/* clearing the array */
 	if (size == 0) {
 		zend_long i;
+		zval *elements = array->elements;
+
+		array->elements = NULL;
 
 		for (i = 0; i < array->size; i++) {
-			zval_ptr_dtor(&(array->elements[i]));
+			zval_ptr_dtor(&(elements[i]));
 		}
 
-		if (array->elements) {
-			efree(array->elements);
-			array->elements = NULL;
+		if (elements) {
+			efree(elements);
```

Comment:
The `if (elements)` isn't strictly necessary here (`efree` accepts NULL).

Reply:
Right. Thanks!